qiskit/ignis/verification/entanglement/parallelize.py

```diff
@@ -0,0 +1,449 @@
+        each item in coupling map), but useful to run anyway
+        '''
+        cmap_new = list(self.cmap)
+        for a in self.cmap:
+            if [a[1], a[0]] not in self.cmap:
+                cmap_new.append([a[1], a[0]])
+        cmap_new.sort(key=lambda x: x[0])
+        return cmap_new
+
+    def get_best_node(self):
+        '''
+        First node with the most connections; Does not yet sort
+        based on error, but that is probably not too useful
+        '''
+
+        bestNode = 0
+        for i in self.nodes:
+            if len(self.nodes[i]) > len(self.nodes[bestNode]):
+                bestNode = i
+            else:
+                pass
+        return bestNode
+
+    def indicator_off(self):
+        '''
+        We turn off gate-based sorting of the tier_dict
+        '''
+
+        self.indicator = False
+
+    def indicator_on(self):
+        '''
+        We turn off gate-based sorting of the tier_dict
+        '''
+
+        self.indicator = True
+
+    def get_cx_error(self):
+        '''
+        Gets dict of relevant CNOT gate errors
+        '''
+
+        a = self.backend.properties().to_dict()['gates']
+        cxerrordict = {}
+        for i in a:
+            if len(i['qubits']) == 1:
+                continue
+            if len(i['qubits']) == 2:
+                b = tuple(i['qubits'])
+                if b in cxerrordict.keys():
+                    pass
+                cxerrordict[b] = i['parameters'][0]['value']
+                if (b[1], b[0]) not in cxerrordict.keys():
+                    cxerrordict[(b[1], b[0])] = i['parameters'][0]['value']
+                else:
+                    pass
+
+        newcxerrordict = cxerrordict
+
+#         for a in list(cxerrordict.keys()):
+#             if (a[1],a[0]) not in cxerrordict.keys():
+#                 newcxerrordict[(a[1],a[0])] = cxerrordict[a]
+        return newcxerrordict
+
+    def get_cx_length(self):
+        '''
+        Gets dict of relevant CNOT gate lengths
+        '''
+
+        a = self.backend.properties().to_dict()['gates']
+        cxlengthdict = {}
+        for i in a:
+            if len(i['qubits']) == 1:
+                continue
+            if len(i['qubits']) == 2:
+                b = tuple(i['qubits'])
+                if b in cxlengthdict.keys():
+                    pass
+                cxlengthdict[b] = i['parameters'][1]['value']
+                if (b[1], b[0]) not in cxlengthdict.keys():
+                    cxlengthdict[(b[1], b[0])] = i['parameters'][1]['value']
+                else:
+                    pass
+
+        newcxlengthdict = cxlengthdict
+#         for a in list(cxlengthdict.keys()):
+#             if (a[1],a[0]) not in cxlengthdict.keys():
+#                 newcxlengthdict[(a[1],a[0])] = cxlengthdict[a]
+
+        return newcxlengthdict
+
+    def child_sorter(self, children, parent):
+        '''
+        Sorts children nodes based on error/length
+        '''
+
+        newchildren = {}
```

Comment:
I'm not sure why this function can't be a one-liner:

`return sorted(children, lambda child: self.get_cx_error()[(child, parent)])

Reply:
done